src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/GraphSymbolUtils.cs

```diff
@@ -143,18 +143,161 @@ private static IEnumerable<INamedTypeSymbol> GetAllViewTypesFromPXGraphOrPXGraph
+                            .TakeWhile(baseGraph => !baseGraph.IsGraphBaseType())
+                            .SelectMany(baseGraph => GetViewDelegatesFromGraphOrGraphExtension(baseGraph, views, pxContext, cancellation));
+            }
+            else
+            {
+                return GetViewDelegatesFromGraphOrGraphExtension(graph, views, pxContext, cancellation);
+            }
+        }
+
+        public static IEnumerable<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)> GetViewDelegatesFromGraphExtensionAndBaseGraph(
+            this ITypeSymbol graphExtension, IEnumerable<ISymbol> views, PXContext pxContext, CancellationToken cancellation)
+        {
+            graphExtension.ThrowOnNull(nameof(graphExtension));
+            views.ThrowOnNull(nameof(views));
+            pxContext.ThrowOnNull(nameof(pxContext));
+
+            return GetViewInfoFromGraphExtension<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)>(
+                graphExtension, pxContext, AddDelegatesFromGraph, AddDelegatesFromGraphExtension);
+
+            void AddDelegatesFromGraph(Dictionary<string, (MethodDeclarationSyntax, IMethodSymbol)> delegates, ITypeSymbol graph)
+            {
+                IEnumerable<(MethodDeclarationSyntax, IMethodSymbol)> dels = graph.GetViewDelegatesFromGraph(views, pxContext, cancellation);
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates.Add(d.Item2.Name, d);
+                }
+            }
+
+            void AddDelegatesFromGraphExtension(Dictionary<string, (MethodDeclarationSyntax, IMethodSymbol)> delegates, ITypeSymbol graphExt)
+            {
+                IEnumerable<(MethodDeclarationSyntax, IMethodSymbol)> dels = graphExt.GetViewDelegatesFromGraphOrGraphExtension(views, pxContext, cancellation);
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates[d.Item2.Name] = d;
+                }
+            }
+        }
+
+        private static IEnumerable<T> GetViewInfoFromGraphExtension<T>(ITypeSymbol graphExtension, PXContext pxContext,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphViewInfo,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphExtensionViewInfo)
+        {
+            IEnumerable<T> empty = Enumerable.Empty<T>();
+
+            if (!graphExtension.InheritsFrom(pxContext.PXGraphExtensionType) || !graphExtension.BaseType.IsGenericType)
+            {
+                return empty;
+            }
+
+            INamedTypeSymbol baseType = graphExtension.BaseType;
+            Dictionary<string, T> infoByView = new Dictionary<string, T>(StringComparer.OrdinalIgnoreCase);
+
+            for (int i = baseType.TypeArguments.Length - 1; i >= 0; i--)
+            {
+                ITypeSymbol argType = baseType.TypeArguments[i];
+
+                if (i == baseType.TypeArguments.Length - 1)
```

Comment:
@metallmonster maybe the last type argument should be analyzed outside of the cycle and start iterations from the  `baseType.TypeArguments.Length - 2`. This will reduce the the complexity of the cycle as well as the number of comparison operations inside.

Reply:
Ok